### DISCUSSION TITLE
running without any paid api keys

### DISCUSSION THREAD

**Discussion by parthwagh9999** [ALwrity/ALwrity#99] [Q&A] [answered]
I am interested in writing blog post that can be generated using open sourced llm models running totally local.
I have stable diffusions also, which can be run locally.

does your tool uses local llm models for generating blogs if available.
Or I have to use Api keys itself.

**Comment by AJaySi** [comment, marked as answer]
Take a look here : https://github.com/AJaySi/AI-Writer/issues/51
ollama will help run models locally. We havent taken up this enhancement as there is not much interest.

For now, if privacy is not a concern, use Google Gemini OR chatgpt 3.5 API key. Both are free and work well with Alwrity.

**Reply by parthwagh9999** [reply]
thank you for quick reply
